apps/frontend/src/lib/psi/refresh.ts

```diff
@@ -0,0 +1,242 @@
+import {
+  tensionPairs,
+  hotTakeLabels
+} from "@/common/constants";
+import { hashCommit } from "@/lib/psi/hash";
+import { BASE_API_URL } from "@/config";
+import { getConnectionSigPubKey } from "@/lib/user";
+import { socketEmit } from "@/lib/socket";
+import {
+  errorToString,
+  SocketRequestType
+} from "@types";
+import {
+  User,
+  Connection,
+  UserData,
+  Session
+} from "@/lib/storage/types";
+import { storage } from "@/lib/storage";
+import { sendMessages } from "@/lib/message";
+import { Socket } from "socket.io-client";
+import { updateUserData } from "@/lib/storage/localStorage/user/userData";
+
+export type Intersection = {
+  tensions: string[];
+  hotTakes: string[];
+  contacts: string[];
+  devconEvents: string[];
+  programmingLangs: string[];
+  starredRepos: string[]
+}
+
+export const refreshPSI = async (user: User, connection: Connection): Promise<Intersection | null> => {
+  try {
+    let tensions: string[] = [];
+    if (user.userData.tensionsRating?.revealAnswers) {
+      const tensionData = tensionPairs.map((tension, index) =>
+        user.userData.tensionsRating!.tensionRating[index] < 50
+          ? tension[0]
+          : tension[1]
+      );
+      tensions = await hashCommit(
+        user.encryptionPrivateKey,
+        connection.user.encryptionPublicKey,
+        tensionData
+      );
+    }
+
+    let hotTakes: string[] = [];
+    if (user.userData.hotTakesRating?.revealAnswers) {
+      const hotTakeData = hotTakeLabels.map((hotTake, index) => {
+        switch (user.userData.hotTakesRating!.rating[index]) {
```

Comment:
Little confused how this works - doesn't there need to be a unique string in the hash commit, such as the name of the hotTake, so that the hashes are actually unique?

Reply:
I'll look into this in the Github PSI PR.